lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -233,13 +256,45 @@ def _check_size():
             return self.size == keysize
 
         self.fingerprint = fingerprint
-        self.public_key = pubkey
 
         if not perms_required:
             return _check_state() and _check_type() and _check_size()
 
         return _check_state() and _check_perms(module) and _check_type() and _check_size()
 
+    def isPublicKeyValid(self, module):
+
+        def _get_pubkey_content():
+            if os.path.exists(self.path + ".pub"):
+                with open(self.path + ".pub", "r") as pubkey_f:
+                    present_pubkey = pubkey_f.read().strip('\n')
+                return present_pubkey
+            else:
+                return False
+
+        def _pubkey_valid(pubkey):
+            if _get_pubkey_content():
+                present_pubkey = _get_pubkey_content()
+                return ' '.join(present_pubkey.split(' ', 2)[0:2]) == pubkey
+            return False
+
+        def _comment_valid():
+            if _get_pubkey_content():
+                present_pubkey = _get_pubkey_content()
+                present_comment = present_pubkey.split(' ')[-1]
```

Comment:
It looks to me like you're assuming that the commen this always there (is it like that? I don't know) and has no spaces in it.

Reply:
That's not exactly the case.
The structure of the public key, standardized.
If comment os provided during the module call, the "_comment_valid" function is called.

The last part of the public key is taken.
If comment exists within the public key, it will always be the last section of the key.

Then, the output compared to the self.comment. If equal, it returns true. If not, false.

Reply:
If you would have `present_pubkey.split(' ', 2)[-1]`, I would agree, but you're calling `present_pubkey.split(' ')[-1]`. So if `present_pubkey` is `abc defgh my comment with spaces`, then `present_pubkey.split(' ', 2)[-1] == "spaces"` and `present_pubkey.split(' ')[-1] == "my comment with spaces"`.

Anyway, please also consider that if `pubkey_present` is `abc defgh`, then both will return `present_comment == "defgh"`, which is not what you want.

Reply:
It might make sense to create an extra function for parsing public keys:
```
def parse_public_key(public_key):
    parts = public_key.split(' ', 2)
    return parts[0], parts[1], '' if len(parts) <= 2 else parts[2]
```

Reply:
Regarding the change of the split, I agree with you.
The "present_pubkey.split(' ', 2)[-1]" is implemented.

Regarding the second part, I can basically create a separate function, but I think that is not required.

For example, the public key missing the comment and the comment is passed to the module.

pubkey_present is abc defg
The result will be False for the

return present_comment == "defgh"

And that could be fine because in such a situation the comment will be readded to the public key in any case.

What do you think?
If you still would like to add the specific function, I could add it.

Reply:
That's true, but once someone starts refactoring the code to support empty comments, this can be a problem. (Also, in the very unlikely case that the comment equals the public key, this could be a problem.)

Finally, it's much better for maintenance if parsing the public key is handled in one place, and not redone in many different cases. So adding a specific function for this is probably making life a lot easier in the future :)

Reply:
Agree.
Will add.